### PR TITLE
ADEN-6123 Fix FMR trigger

### PR DESCRIPTION
**Pull request by v3rth**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java [hunk 1]
```diff
@@ -601,6 +606,31 @@ public AdsBaseObject triggerAdSlot(String slotName) {
     return this;
   }
 
+  private void triggerFMR() {
+    final By FMRSelector = By.cssSelector(AdsContent.getSlotSelector(AdsContent.FLOATING_MEDREC));
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: MAJOR Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00117)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java [hunk 2]
```diff
@@ -601,6 +606,31 @@ public AdsBaseObject triggerAdSlot(String slotName) {
...
+  }
+
+  private void doUntilElementVisible(By by, Runnable f) {
+    Boolean isElementDisplayed = false;
+    changeImplicitWait(0, TimeUnit.MILLISECONDS);
+
+    do {
+      try {
+        isElementDisplayed = driver.findElement(by).isDisplayed();
+      } catch (NoSuchElementException ignored) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: MAJOR Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L649](https://github.com/Wikia/selenium-tests/blob/6a4db2cfa50a79ca6a8bcf2c92f4b0a09be85d27/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L649): Introduce a new variable instead of reusing the parameter "duration". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L649](https://github.com/Wikia/selenium-tests/blob/a610a3bb1da92877571951d9c52347847b9e5d5a/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L649): Introduce a new variable instead of reusing the parameter "duration". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdsBaseObject.java#L648](https://github.com/Wikia/selenium-tests/blob/2cf3e111c72f272e165b379aa0cf7674536c5b52/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java#L648): Introduce a new variable instead of reusing the parameter "duration". [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1226)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java [hunk 3]
```diff
@@ -601,6 +606,30 @@ public AdsBaseObject triggerAdSlot(String slotName) {
     return this;
   }
 
+  private void triggerFMR() {
+    scrollToPosition(By.cssSelector("#wikia-recent-activity"));
+
+    doUntilElementVisible(By.cssSelector(AdsContent.getSlotSelector(AdsContent.FLOATING_MEDREC)), () -> {
+      jsActions.scrollBy(0, 100);
+      wait.forX(Duration.ofSeconds(1));
```
Comment (bulaszenko) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: If we will have defect like lack of FLOATING_MEDREC test never ends we should implement like check max 5 times

Comment (v3rth) [reply src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java]: Good point! Thanks :)

**Pull request merged**